WordPress/Classes/Utility/Ratings/AppRatingUtility.m

```diff
@@ -53,23 +112,57 @@ + (void)initializeForVersion:(NSString *)version
     else
     {
         // Restarting tracking for new version of app
-        
         BOOL interactedWithAppReviewPromptInPreviousVersion = [self interactedWithAppReviewPrompt];
         
         [userDefaults setObject:version forKey:AppRatingCurrentVersion];
         [userDefaults setInteger:0 forKey:AppRatingSignificantEventCount];
+        for (NSString *section in [AppRatingUtility sharedInstance].sections.allKeys) {
+            [userDefaults setInteger:0 forKey:[self significantEventCountKeyForSection:section]];
+        }
         [userDefaults setBool:NO forKey:AppRatingRatedCurrentVersion];
         [userDefaults setBool:NO forKey:AppRatingDeclinedToRateCurrentVersion];
         [userDefaults setBool:NO forKey:AppRatingGaveFeedbackForCurrentVersion];
+        [userDefaults setBool:NO forKey:AppRatingDislikedCurrentVersion];
+        [userDefaults setBool:NO forKey:AppRatingLikedCurrentVersion];
         
-        // Note - this is a temporary fix for 4.6.1 so we don't double prompt users
-        // for a review within the span of a few days. Make sure to remove this for 4.7
-        if ([trackingVersion isEqualToString:@"4.6"] && interactedWithAppReviewPromptInPreviousVersion) {
-            [userDefaults setBool:YES forKey:AppRatingRatedCurrentVersion];
+        [self resetReviewPromptDisabledStatus];
+        
+        if (interactedWithAppReviewPromptInPreviousVersion) {
+            NSInteger numberOfVersionsSkippedPrompting = [userDefaults integerForKey:AppRatingNumberOfVersionsSkippedPrompting];
+            NSInteger numberOfVersionsToSkipPrompting = [userDefaults integerForKey:AppRatingNumberOfVersionsToSkipPrompting];
+            
+            if (numberOfVersionsToSkipPrompting > 0) {
+                if (numberOfVersionsSkippedPrompting < numberOfVersionsToSkipPrompting) {
+                    // We haven't skipped enough versions, skip this one
+                    numberOfVersionsSkippedPrompting++;
+                    [userDefaults setInteger:numberOfVersionsSkippedPrompting forKey:AppRatingNumberOfVersionsSkippedPrompting];
+                    [userDefaults setBool:YES forKey:AppRatingRatedCurrentVersion];
```

Comment:
I just realized that the reason this is set to YES is probably to trigger the `interactedWithAppReviewPromptInPreviousVersion` condition above on the next version of the app.  Is this correct?

If so, I'd consider leaving this as NO (since the user didn't really rate the current version, and it could be hard to read for other developers, and instead replace the following line above:

`if (interactedWithAppReviewPromptInPreviousVersion) {`

...with something like...

`if ([userDefaults integerForKey:AppRatingNumberOfVersionsToSkipPrompting] > 0) {`